### PR TITLE
=doc #19551 Supplement java8-lambda docs of persistence stash

### PR DESCRIPTION
**Pull request by MQ-EL**
**Sorry** for missing that and fix wrong msg info of stashoverflowException and docs typo. :(
refs: pr #19626 , issue #19551

### CODE REVIEW

**Comment by patriknw** [comment]
OK TO TEST

**Comment by patriknw** [comment]
LGTM, thanks for following up, @MQ-EL

File: akka-docs/rst/scala/persistence.rst [hunk 1]
```diff
@@ -214,8 +212,8 @@ You can also query default strategy via the Akka persistence extension singleton
     Persistence(context.system).defaultInternalStashOverflowStrategy
 
 .. note::
-  The bounded mailbox should be avoid in the persistent actor, because it may be discarding the messages come from 
-  Storage backends. You can use bounded stash instead of bounded mailbox.
+  The bounded mailbox should be avoid in the persistent actor, by which the messages come from Storage backends may
```
Comment (johanandren) [inline akka-docs/rst/scala/persistence.rst]: avoid*ed*?

File: akka-docs/rst/java/persistence.rst [hunk 2]
```diff
@@ -228,11 +226,11 @@ The ``DiscardToDeadLetterStrategy`` strategy also has a pre-packaged companion c
...
 You can also query default strategy via the Akka persistence extension singleton::
 
-    Persistence.get(context().system()).defaultInternalStashOverflowStrategy()
+    Persistence.get(context().system()).defaultInternalStashOverflowStrategy();
 
 .. note::
-  The bounded mailbox should be avoid in the persistent actor, because it may be discarding the messages come from 
-  Storage backends. You can use bounded stash instead of bounded mailbox.
+  The bounded mailbox should be avoid in the persistent actor, by which the messages come from Storage backends may 
+  be discarded. You can use bounded stash instead of it.
```
Comment (johanandren) [inline akka-docs/rst/java/persistence.rst]: Storage should start with lowercase s?

**Comment by johanandren** [comment]
LGTM (aside from nitpicks that was in the original text rather than introduced by the PR)

**Comment by johanandren** [comment]
Test faillure is unrelated.

File: akka-docs/rst/java/lambda-persistence.rst [hunk 3]
```diff
@@ -202,6 +190,46 @@ and before any other received messages.
...
 is called (logging the error by default), and the actor will be stopped.
 
+.. _internal-stash-lambda:
+
+Internal stash 
+--------------
+
+The persistent actor has a private :ref:`stash <stash-lambda>` for internally caching incoming messages during 
+:ref:`recovery <recovery-java-lambda>` or the ``persist\persistAll`` method persisting events. However You can use 
+inherited stash. The internal stash doesn't interfere with it apart from inherited ``unstashAll`` method, which
```
Comment (drewhk) [inline akka-docs/rst/java/lambda-persistence.rst]: "you can still use/inherit from the Stash interface"
"The internal stash cooperates with the normal Stash by hooking into unstashAll and making sure messages are unstashed properly to the internal stash to maintain ordering guarantees."

File: akka-docs/rst/java/lambda-persistence.rst [hunk 4]
```diff
@@ -202,6 +190,46 @@ and before any other received messages.
...
+Internal stash 
+--------------
+
+The persistent actor has a private :ref:`stash <stash-lambda>` for internally caching incoming messages during 
+:ref:`recovery <recovery-java-lambda>` or the ``persist\persistAll`` method persisting events. However You can use 
+inherited stash. The internal stash doesn't interfere with it apart from inherited ``unstashAll`` method, which 
+prepends all the messages in the inherited stash to the internal stash instead of mailbox. 
+
+You should be careful to not send more messages to a persistent actor than it can keep up with, otherwise the number 
+of stashed messages will grow. It can be wise to protect against `OutOfMemoryError` by defining a maximum stash
```
Comment (drewhk) [inline akka-docs/rst/java/lambda-persistence.rst]: "will grow without bounds"

**Comment by drewhk** [comment]
small nitpicks, otherwise LGTM

File: akka-docs/rst/java/lambda-persistence.rst (hunk 4 above)
Comment (ktoso) [inline akka-docs/rst/java/lambda-persistence.rst]: restructured text syntax is:

```
``OutOfMemoryError``
```

File: akka-docs/rst/scala/persistence.rst (hunk 1 above)
Comment (ktoso) [inline akka-docs/rst/scala/persistence.rst]: and lowercase storage here too

File: akka-docs/rst/java/lambda-persistence.rst [hunk 5]
```diff
@@ -202,6 +190,46 @@ and before any other received messages.
 If there is a problem with recovering the state of the actor from the journal, ``onRecoveryFailure``
 is called (logging the error by default), and the actor will be stopped.
 
+.. _internal-stash-lambda:
+
+Internal stash 
+--------------
+
+The persistent actor has a private :ref:`stash <stash-lambda>` for internally caching incoming messages during 
+:ref:`recovery <recovery-java-lambda>` or the ``persist\persistAll`` method persisting events. However You can use
```
Comment (ktoso) [inline akka-docs/rst/java/lambda-persistence.rst]: lowercase `you` please

**Comment by ktoso** [comment]
LGTM after minor comments addressed

**Comment by MQ-EL** [comment]
Thanks :)

File: akka-docs/rst/scala/code/docs/persistence/PersistenceDocSpec.scala [hunk 6]
```diff
@@ -414,51 +414,6 @@ object PersistenceDocSpec {
...
-    // a-inner-2
-    // b-inner-1
-    // b-inner-2
-
-    // which can be seen as the following causal relationship:
-    // a -> a-outer-1 -> a-outer-2 -> a-inner-1 -> a-inner-2
-    // b -> b-outer-1 -> b-outer-2 -> b-inner-1 -> b-inner-2
-
-    //#nested-persistAsync-persistAsync-caller
   }
```
Comment (MQ-EL) [inline akka-docs/rst/scala/code/docs/persistence/PersistenceDocSpec.scala]: redundance

Comment (ktoso) [inline akka-docs/rst/scala/code/docs/persistence/PersistenceDocSpec.scala]: How come? They're used in `persistence.rst`

```
It is also possible to nest ``persistAsync`` calls, using the same pattern:

.. includecode:: code/docs/persistence/PersistenceDocSpec.scala#nested-persistAsync-persistAsync
```

Comment (MQ-EL) [inline akka-docs/rst/scala/code/docs/persistence/PersistenceDocSpec.scala]: Already exists in [PersistenceDocSpec$NestedPersists]( https://github.com/MQ-EL/akka/blob/master/akka-docs/rst/scala/code/docs/persistence/PersistenceDocSpec.scala#L325.
).

Comment (ktoso) [inline akka-docs/rst/scala/code/docs/persistence/PersistenceDocSpec.scala]: Oh wow, I see - seems we had these duplicated due to some merge gone weird, thanks!.

**Pull request merged**